06_puppy/n0npax/syncStore.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"errors"
+)
+
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+func (m *SyncStore) CreatePuppy(puppy *Puppy) int {
+	var newID int
+	m.Range(func(_, _ interface{}) bool {
+		newID++
+		return true
+	})
+	puppy.ID = newID
+	m.Store(puppy.ID, *puppy)
+	return puppy.ID
+}
+
+func (m *SyncStore) ReadPuppy(id int) (*Puppy, error) {
+	puppyData, ok := m.Load(id)
+	if !ok {
+		return nil, errors.New("doesn't exists")
+	}
+	puppy := puppyData.(Puppy)
+	return &puppy, nil
+}
+
+func (m *SyncStore) UpdatePuppy(id int, puppy *Puppy) error {
```

Comment:
Again here, I think there should be a check to see if the id exists, and if not a suitable error condition

Reply:
fixed. Thanks!